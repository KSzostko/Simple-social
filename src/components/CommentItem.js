import React from 'react';
import styled from 'styled-components';

const StyledListItem = styled.li`
    border: 1px solid #000;
    padding:  20px 10px;
    margin-bottom: 20px;
`;

const StyledHeader = styled.header`
    display: flex;
    justify-content: space-between;
`;

const StyledTitle = styled.h3`
    font-size: 0.9rem;
    margin: 0;
`;

const StyledEmail = styled.span`
    font-size: 0.8rem;
    color: #0000ee;
    text-decoration: underline;
    cursor: pointer;
`;

const StyledParagraph = styled.p`
    font-size: 0.9rem;
    margin-bottom: 0;
`;

function CommentItem({ data }) {
    const { name, email, body } = data;
    
    return (
        <StyledListItem>
            <article>
                <StyledHeader>
                    <StyledTitle>{name}</StyledTitle>
                    <StyledEmail>{email}</StyledEmail>
                </StyledHeader>
                <StyledParagraph>{body}</StyledParagraph>
            </article>
        </StyledListItem>
    );
}

export default CommentItem;