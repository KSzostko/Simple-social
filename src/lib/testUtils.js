const fakePost = () => ({
    userId: 1,
    postId: 2,
    title: 'Post title',
    body: 'Post body',
});

const fakeComment = () => ({
    postId: 1,
    id: 1,
    name: 'Comment name',
    email: 'user@gmail.com',
    body: 'Comment body',
});

export { fakePost, fakeComment };