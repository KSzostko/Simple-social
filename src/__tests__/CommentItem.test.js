import React from 'react';
import { render } from '@testing-library/react';
import { fakeComment } from '../lib/testUtils';
import CommentItem from '../components/CommentItem';

const fakeData = {
    data: { ...fakeComment() }
};

describe('<CommentItem />', () => {
    it('renders and matches the snapshot', () => {
        const { container } = render(<CommentItem {...fakeData} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});