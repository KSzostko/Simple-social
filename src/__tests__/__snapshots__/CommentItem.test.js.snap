// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentItem /> renders and matches the snapshot 1`] = `
<li
  class="sc-AxjAm iAtNxT"
>
  <article>
    <header
      class="sc-AxirZ lcgpFI"
    >
      <h3
        class="sc-AxiKw cckNeg"
      >
        Comment name
      </h3>
      <span
        class="sc-AxhCb hcObeU"
      >
        user@gmail.com
      </span>
    </header>
    <p
      class="sc-AxhUy hRwRRj"
    >
      Comment body
    </p>
  </article>
</li>
`;
